server/website/website/db/base/parser.py

```diff
@@ -99,10 +106,11 @@ def valid_boolean_val_to_string(self):
             str_false += str(bval) + ' '
         return str_true + '; ' + str_false
 
-    def convert_dbms_knobs(self, knobs):
+    def convert_dbms_knobs(self, knobs, knob_catalog=None):
         knob_data = {}
-        tunable_knob_catalog = KnobCatalog.objects.filter(dbms__id=self.dbms_id, tunable=True)
-        for metadata in tunable_knob_catalog:
+        if knob_catalog is None:
+            knob_catalog = KnobCatalog.objects.filter(dbms__id=self.dbms_id, tunable=True)
```

Comment:
If we are deprecating the tunable field in the KnobCatalog then shouldn't we be initializing `knob_catalog` with the SessionKnob objects? Or at least without the `tunable=True` arg?

Reply:
It is not necessary to take action about the tunable field now. It can be used as a hard limitation of session tunable knobs (but now it is changed according to the change of session tunable knobs). We can keep an eye on that field and if in the future it is proved to be totally useless, we can deprecate it.